### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by Pavellys**
added description for Test, @Step, logs, circleCI, allure-report

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            - ~/.m2
+          key: v1-dependencies-{{ checksum "pom.xml" }}
+
+      # run tests!
+      - run:
+          name: Get chrome version
+          command: google-chrome --version
+      #      - run:
+      #          name: Setting chmod for Chrome
+      #          command: chmod 777 src/test/resources/webdrivers/linux/chromedriver
```
Comment (vchesnakoutest) [inline .circleci/config.yml:31]: убрать

File: pom.xml [hunk 2]
```diff
@@ -19,6 +19,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.0</version>
```
Comment (vchesnakoutest) [inline pom.xml:22]: вынести в properties

File: src/test/java/tests/CartTest.java [hunk 3]
```diff
@@ -5,44 +5,44 @@
...
     public void addProductInCartTest() {
         cartPage.openPage(PROMOTIONS_URL)
                 .clickOnProduct(PROMOTION_PRODUCT)
-                .clickToButtonInCart()
-                .isPopUpDisplayed()
+                .clickOnButtonInCart()
+                .waitForCartPopUpDisplayed()
                 .openPage(CART_URL);
-        Assert.assertTrue(cartPage.checkInCart(PROMOTION_PRODUCT));
+        Assert.assertTrue(cartPage.isCheckInCart(PROMOTION_PRODUCT));
```
Comment (vchesnakoutest) [inline src/test/java/tests/CartTest.java:15]: isProductIsAddedToCart

File: src/main/java/pages/CartPage.java [hunk 4]
```diff
@@ -30,67 +33,93 @@ public CartPage(WebDriver driver) {
...
     public CartPage clickCheckboxDeliveryOnRB(){
+        log.info("Clicking in checkbox 'Доставка по РБ'");
         checkboxDeliveryOnRB.click();
         return this;
     }
 
-    public CartPage inputAddressUserData(AddressUser addressUser){
+    @Step("Entering data in address form 'Доставка по РБ'")
+    public CartPage inputAddressUserDataInAddressField(AddressUser addressUser){
+        log.info("Entering data in address form 'Доставка по РБ'");
```
Comment (vchesnakoutest) [inline src/main/java/pages/CartPage.java:122]: String.format("Entering address: '%s' in address form 'Доставка по РБ'", addressUser)

userAddress

**Review by vchesnakoutest — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (Pavellys) [reply .circleci/config.yml:31]: done

File: src/test/java/tests/CartTest.java (hunk 3 above)
Comment (Pavellys) [reply src/test/java/tests/CartTest.java:15]: done

File: src/main/java/pages/CartPage.java (hunk 4 above)
Comment (Pavellys) [reply src/main/java/pages/CartPage.java:122]: done

File: pom.xml (hunk 2 above)
Comment (Pavellys) [reply pom.xml:22]: done

**Pull request merged**